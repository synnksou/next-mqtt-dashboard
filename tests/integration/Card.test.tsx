import { render } from "@testing-library/react"

import {
  Card,
  CardContent,
  CardDescription,
  CardFooter,
  CardHeader,
  CardTitle,
} from "@/components/ui/card"

describe("Card Component", () => {
  it("renders Card correctly", () => {
    const { asFragment } = render(
      <Card>
        <CardHeader>
          <CardTitle>Test Title</CardTitle>
          <CardDescription>Test Description</CardDescription>
        </CardHeader>
        <CardContent>Test Content</CardContent>
        <CardFooter>Test Footer</CardFooter>
      </Card>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
