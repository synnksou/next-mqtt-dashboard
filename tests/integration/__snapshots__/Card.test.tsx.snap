// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card Component > renders Card correctly 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <h3
        class="text-lg font-semibold leading-none tracking-tight"
      >
        Test Title
      </h3>
      <p
        class="text-sm text-muted-foreground"
      >
        Test Description
      </p>
    </div>
    <div
      class="p-6 pt-0"
    >
      Test Content
    </div>
    <div
      class="flex items-center p-6 pt-0"
    >
      Test Footer
    </div>
  </div>
</DocumentFragment>
`;
